src/components/checkbox/checkbox.js

```diff
@@ -169,19 +169,29 @@ function MdCheckboxDirective(inputDirective, $mdAria, $mdConstant, $mdTheming, $
         }
       }
 
+      /**
+       * @param {KeyboardEvent} ev 'keypress' event to handle
+       */
       function keypressHandler(ev) {
         var keyCode = ev.which || ev.keyCode;
+        var submit, form;
+
         ev.preventDefault();
         switch (keyCode) {
           case $mdConstant.KEY_CODE.SPACE:
             element.addClass('md-focused');
             listener(ev);
             break;
           case $mdConstant.KEY_CODE.ENTER:
-            var form = $mdUtil.getClosest(ev.target, 'form');
-            // We have to use a native event here as the form directive does not use jqlite
+            // Match the behavior of the native input type="checkbox".
+            // It's important that MouseEvent is used here versus Event in order to match the native
+            // behavior and properly trigger ng-submit.
+            form = $mdUtil.getClosest(ev.target, 'form');
             if (form) {
-              form.dispatchEvent(new Event('submit'));
+              submit = form.querySelector('button[type="submit"]:enabled, input[type="submit"]:enabled');
+              if (submit) {
+                submit.click();
+              }
```

Comment:
I would imagine this would work similarly to inputs.
If a submit button is found, fire a click event onto it, otherwise just submit the form.

This behavior is currently present on inputs: [Code Pen](https://codepen.io/codymikol/pen/OdaQYy?editors=1010#0)

I leverage this kind of behavior in my application. In some circumstances I have an outer controller that doesn't know about it's contents, but has the ability to fire submit events to child forms via a directive. I also want the user to be able to use the keyboard inside of the form to submit.

There is an attribute "form" which allows you to specify the form for a submit button outside of a form, but it is not supported in IE 11.

Reply:
I do see that there is definitely native inconsistency with this. Inputs will submit regardless of submit button presence, but checkboxes require one. However, I don't see any drawback to allowing this kind of behavior in absence of a submit button.

Reply:
Well the primary thing that we have to fix is that `Enter` on checkboxes should not submit the form if the submit button/input is disabled.

I forked your [CodePen](https://codepen.io/Splaktar/pen/NooaYZ?editors=1010#0) and added a native `input`. It submits on `Enter` without a submit button/input *only* if it's the only input in the form. If there are 2 `inputs`, the `Enter` key does not submit without a submit button/input.

Some of this odd behavior is captured in the [AngularJS Form Directive docs](https://docs.angularjs.org/api/ng/directive/form#submitting-a-form-and-preventing-the-default-action).

I suppose we could theoretically submit the form on `Enter` for a checkbox if no submit button or input is found, but is inconsistent with [native behavior](https://jsbin.com/lajutej/edit?html,js,output). So it's likely not a good idea. If you comment out the input in this JSBin, `Enter` on a lone `checkbox` does not submit the form.